homeassistant/components/splunk/__init__.py

```diff
@@ -75,16 +63,53 @@ def setup(hass, config):
     name = conf.get(CONF_NAME)
     entity_filter = conf[CONF_FILTER]
 
-    if use_ssl:
-        uri_scheme = "https://"
-    else:
-        uri_scheme = "http://"
+    if verify_ssl and not use_ssl:
+        _LOGGER.error("You cannot use verify_ssl without ssl")
+        return False
+
+    event_collector = SplunkSender(
+        host=host,
+        port=port,
+        token=token,
+        hostname=name,
+        protocol=["http", "https"][use_ssl],
+        verify=verify_ssl,
+        api_url=SPLUNK_ENDPOINT,
+    )
+
+    batch = []
+
+    def queue(payload):
+        batch.append(json.dumps(payload, cls=JSONEncoder))
+
+    payload = {
+        "time": time.time(),
+        "host": name,
+        "event": {
+            "domain": DOMAIN,
+            "meta": "Home Assistant has started",
+            "value": 1,
+        },
+    }
 
-    event_collector = f"{uri_scheme}{host}:{port}/services/collector/event"
-    headers = {AUTHORIZATION: f"Splunk {token}"}
+    try:
+        event_collector._send_to_splunk("send-event", json.dumps(payload))
```

Comment:
The library enforces additional fields to be sent such as index, source, and sourcetype. Typically these are defined on the Splunk side, and are not currently set in the integration. Adding these to the schema would make this a breaking change. I attempted to leave them blank, but it caused the library to send literally blank fields which Splunk rejected.

This is my frustration with using Pypi libraries rather than building a solution perfectly for Home Assistant.

Reply:
Ok.

Reply:
We can disable the pylint error locally for `protected-access`.